### PR TITLE
Data Models 🎉 🚀 ✔️ 

### PR DESCRIPTION
**Pull request by Nizamuddin4493**
# Data Models 🎉 🚀 ✔️ 
- Installed Ruby on Rails framework.
- Used Ruby Gems as a software packages system.
- Understood what ORM is.
- Used database migration files to maintain database schema.

### CODE REVIEW

**Review by FrederickMih — COMMENTED** [review]
## Hi 🙋  @Nizamuddin4493 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

@Nizamuddin4493 we are expected to use the new and updated [README template](https://github.com/microverseinc/readme-template/edit/master/README.md) for all our projects.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by TedLivist** [comment]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @Nizamuddin4493,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly format your README correctly using the link shared with you previously. Your README should contain the Table of Contents with all the links present and their corresponding sections.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/migrate/20230221081006_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateUsers < ActiveRecord::Migration[7.0]
+  def change
+    create_table :users do |t|
+      t.string :name
+      t.string :photo
+      t.string :bio
+
+      t.timestamps
```
Comment (franclobo) [inline db/migrate/20230221081006_create_users.rb:8]: - [ ] Please 🙏🏽 add `t.integer :post_counter` column to register the number of posts by user.

File: db/migrate/20230221082231_create_posts.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :text
+
+      t.timestamps
```
Comment (franclobo) [inline db/migrate/20230221082231_create_posts.rb:7]: - [ ] Please 🙏🏽 add` t.integer :likes_counter` and `t.integer :comments_counter` columns on Post table to register the number of likes and comments.
- [ ] Kindly add the references to the user table 👉🏽 `t.references :author, foreign_key: { to_table: 'users' }`. It represents the relationship between the user and the posts tables.

File: db/migrate/20230221082102_create_comments.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.text :text
+
+      t.timestamps
```
Comment (franclobo) [inline db/migrate/20230221082102_create_comments.rb:6]: - [ ] Please 🙏🏽 add the references to the user and the post tables. It represents the relationship between these tables and the comments tables.

File: db/migrate/20230221082316_create_likes.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+
+      t.timestamps
```
Comment (franclobo) [inline db/migrate/20230221082316_create_likes.rb:5]: - [ ] Please 🙏🏽 add the references to the user and the post tables. It represents the relationship between these tables and the like table.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Nizamuddin4493, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- No linter errors ✔️ 
- Correct Git flow ✔️ 
- Postgresql is used 💯 
- migrations are created 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- **[OPTIONAL]** Please 🙏🏽 add the LICENSE file to your repo and make sure that the MIT link works as expected for a professional README file.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
**Hi @Nizamuddin4493 
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights

✔️ No linters error
✔️  Well done implementing the requested changes 
## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**